rskj-core/src/main/java/co/rsk/net/discovery/message/MessageDecoder.java

```diff
@@ -48,7 +49,7 @@ public static PeerDiscoveryMessage decode(byte[] wire) {
         byte[] data = new byte[wire.length - 98];
         System.arraycopy(wire, 98, data, 0, data.length);
 
-        byte[] mdcCheck = sha3(wire, 32, wire.length - 32);
+        byte[] mdcCheck = HashUtil.keccak256(wire, 32, wire.length - 32);
```

Comment:
Don't we want the below comparison to be done with the class compareTo() method?